### PR TITLE
[FINERACT-1941] change task executors to threadpool task executor

### PR DESCRIPTION
**Pull request by taskain7**
## Description

Change task executors to threadpool task executor.
Make threadpool task executors configurable via environment properties.

### CODE REVIEW

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/core/config/FineractProperties.java [hunk 1]
```diff
@@ -190,6 +192,30 @@ public static class FineractEventsProperties {
         private FineractExternalEventsProperties external;
     }
 
+    @Getter
+    @Setter
+    public static class FineractTaskExecutor {
+
+        private FineractEventTaskExecutor eventTaskExecutor;
```
Comment (galovics) [inline fineract-core/src/main/java/org/apache/fineract/infrastructure/core/config/FineractProperties.java]: I don't think this is in the right place. The property organization was horizontal until this point and now you're making it vertical. 

The event related configuration should be with the rest of the event properties.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/jobs/sendmessagetosmsgateway/SendMessageToSmsGatewayConfig.java [hunk 2]
```diff
@@ -46,6 +48,9 @@ public class SendMessageToSmsGatewayConfig {
     private NotificationSenderService notificationSenderService;
     @Autowired
     private SmsConfigUtils smsConfigUtils;
+    @Autowired
+    @Qualifier("fineractDefaultThreadPoolTaskExecutor")
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/jobs/sendmessagetosmsgateway/SendMessageToSmsGatewayConfig.java]: I think you could create a constant on these specific bean name(s) so you don't have to put this string value everywhere.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/jobs/recalculateinterestforloan/RecalculateInterestForLoanTasklet.java [hunk 3]
```diff
@@ -91,8 +91,8 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
 
     private void recalculateInterest(OfficeData office, int threadPoolSize, int batchSize) {
         final int pageSize = batchSize * threadPoolSize;
-
-        final ExecutorService executorService = Executors.newFixedThreadPool(threadPoolSize);
+        taskExecutor.setCorePoolSize(threadPoolSize);
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/jobs/recalculateinterestforloan/RecalculateInterestForLoanTasklet.java:97]: Sooo, are we updating the global default task executor configuration? This mustn't be right..

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/jobs/postinterestforsavings/PostInterestForSavingTasklet.java [hunk 4]
```diff
@@ -55,15 +55,18 @@ public class PostInterestForSavingTasklet implements Tasklet {
...
     private final Queue<List<SavingsAccountData>> queue = new ArrayDeque<>();
     private final ApplicationContext applicationContext;
+    @Qualifier("fineractDefaultThreadPoolTaskExecutor")
+    private final ThreadPoolTaskExecutor taskExecutor;
     private final int queueSize = 1;
 
     @Override
     public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
         final int threadPoolSize = Integer.parseInt((String) chunkContext.getStepContext().getJobParameters().get("thread-pool-size"));
+        taskExecutor.setCorePoolSize(threadPoolSize);
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/jobs/postinterestforsavings/PostInterestForSavingTasklet.java:66]: Same as for the other one. This shouldn't be done at all.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/core/config/FineractProperties.java (hunk 1 above)
Comment (taskain7) [reply fineract-core/src/main/java/org/apache/fineract/infrastructure/core/config/FineractProperties.java]: replaced under the event properties

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/jobs/sendmessagetosmsgateway/SendMessageToSmsGatewayConfig.java (hunk 2 above)
Comment (taskain7) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/jobs/sendmessagetosmsgateway/SendMessageToSmsGatewayConfig.java]: constants are used now

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/jobs/recalculateinterestforloan/RecalculateInterestForLoanTasklet.java (hunk 3 above)
Comment (taskain7) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/jobs/recalculateinterestforloan/RecalculateInterestForLoanTasklet.java:97]: a configurable task executor has been made, which is a prototype bean

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/jobs/postinterestforsavings/PostInterestForSavingTasklet.java (hunk 4 above)
Comment (taskain7) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/jobs/postinterestforsavings/PostInterestForSavingTasklet.java:66]: a configurable task executor has been made, which is a prototype bean

**Review by galovics — APPROVED** [review]

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**